### DISCUSSION TITLE
How to control when to prompt

### DISCUSSION THREAD

**Discussion by Yingqi39** [jspsych/jsPsych#728] [Q&A]
Hello,

I'm new to both js and jspsych.

For my experiment, I just wonder how I can control when to prompt my html texts for I want it to appear at random time during the audio trial.

```js
  var left_side_key='f';
  var right_side_key='j';

  var timeline = [];
  
  var trial = {
      type: "audio-keyboard-response",
      stimulus: "sound/tone.mp3",
      choices: ['f', 'j'],
      prompt: "<p>Is the pitch high or low? Press 'f' for low and 'j' for high.</p>",
      trial_ends_after_audio: true
  };  

  timeline.push(trial);

	jsPsych.init({
	  timeline: timeline,
    use_webaudio: false
	});
```
Thank you.

**Comment by jodeleeuw** [comment]
Quick clarification questions: you want the audio to start playing, and then some random time later have the prompt appear? Is there a minimum gap that you want? How long is the sound?

**Reply by Yingqi39** [reply]
My sound continues for 8 seconds. During this 8-second trail, a html texts will appear at some time and stay until the sound ends. I haven't holden a specific idea on the span of random time. For later experiment, I want to make participants to have different time for thinking of a more complex html text.

**Reply by jodeleeuw** [reply]
You'll probably need to create a modified version of the plugin to do this. There's no parameter right now that delays the appearance of the prompt. Normally you could break this up into two consecutive trials, where the second trial has the prompt and the first one doesn't, but since your audio file continues throughout this isn't possible.
